### DISCUSSION TITLE
uiSlider LinearBarVertical setSliderStyle

### DISCUSSION THREAD

**Discussion by dnaldoog** [RomanKubiak/ctrlr#627] [Show and tell]
Here's how to create vertical linear bars:

`panel:getSlider("lfo rate"):getOwnedSlider():setSliderStyle(Slider.LinearBarVertical)`

on panel load. The trick here is to set `Value background colour` to `00000000`

I don't think this can be set in the editor as far as I can see. Linear Bar always goes left to right.

----
![linearbarvertical](https://github.com/RomanKubiak/ctrlr/assets/76831683/e83bdc63-5fab-48f1-8126-6eee238118ec)

[Set Linear Bar Vertical_1.zip](https://github.com/RomanKubiak/ctrlr/files/14731359/Set.Linear.Bar.Vertical_1.zip). 

Credits: @ted-juh

**Comment by damiensellier** [comment]
Some types are definitely missing in our ctrlr uiSlider types 
I'll add them in a future update.

LinearHorizontal  **OK**
LinearVertical **OK**
LinearBar **OK**
Rotary **OK**
RotaryHorizontalDrag **OK**
RotaryVerticalDrag **OK**
IncDecButtons **OK**

LinearBarVertical **MISSING**
RotaryHorizontalVerticalDrag **MISSING**
TwoValueHorizontal **MISSING**
TwoValueVertical **MISSING**
ThreeValueHorizontal **MISSING**
ThreeValueVertical **MISSING**

https://docs.juce.com/master/classSlider.html#af1caee82552143dd9ff0fc9f0cdc0888a8a942813d4d457e0344e725bd9faffc5

**Reply by dnaldoog** [reply]
Excellent! LinearBarVertical looks nice in panels.

**Reply by Ted-juh** [reply]
I tried to add rotaryHoirzontalVerticalDrag yesterday. All the switches were set up correctly. The enums are there. But to no avail and I don't understand why. Going over the ones already added to Ctrlr, I couldn't find anything else to add/ change. Even a clean rebuild didn't add the rotaryHorizontalVerticalDrag to the combobox to be selectable.

So, in short, I hope Damien succeeds.

**Comment by dervur** [comment]
Hi, there I just wanted to say that if you want to add it in your build here is how to do it: https://github.com/RomanKubiak/ctrlr/discussions/703
With this you don't need lua and you can just select it from the dropdown menu.
